src/lib/Folder.interface.ts

```diff
@@ -0,0 +1,401 @@
+  FIDEM = 'FIDEM',
+  FINANCO = 'FINANCO',
+  FRANFINANCE = 'FRANFINANCE',
+  MY_MONEY_BANK = 'MY_MONEY_BANK',
+  CARREFOUR_BANK = 'CARREFOUR_BANK',
+  SOCRAM = 'SOCRAM',
+  SOFINCO = 'SOFINCO',
+  VIAXEL = 'VIAXEL',
+  OTHER = 'OTHER',
+}
+
+/**
+ * File
+ */
+export interface File {
+  createdAt: Date;
+  id: string;
+  internalComments: InternalComment[];
+  metadata: { extension: string; mimeType: string; name: string; size: number };
+  properties: DocumentProperties;
+  rejectionCode: number;
+  signedUrl: string;
+  signedUrlCreatedAt: Date;
+  signedUrlTTL: number;
+  state: FileState;
+  updatedAt: Date;
+}
+
+/**
+ * Legal file
+ */
+export interface LegalFile extends File {
+  type: LegalFileType;
+}
+
+/**
+ * Supporting file
+ */
+export interface SupportingFile extends File {
+  type: SupportingFileType;
+}
+
+/**
+ * State of the file
+ */
+export enum FileState {
+  IN_PROGRESS = 1,
+  REJECTED,
+  SUSPECTED,
+  OK,
+}
+
+/**
+ * Type of the file
+ */
+export enum FileType {
+  CONTRACT = 'CONTRACT',
+  INFORMATION = 'INFORMATION',
+  CONVENTION_OF_PROOF = 'CONVENTION_OF_PROOF',
+  LOAN_AND_PAYMENT_AGREEMENT = 'LOAN_AND_PAYMENT_AGREEMENT',
+  TERMS_OF_SALE = 'TERMS_OF_SALE',
+  SEPA = 'SEPA',
+  BANK_RECORDS = 'BANK_RECORDS',
+  LEGAL_MENTIONS = 'LEGAL_MENTIONS',
+  RECEIPT = 'RECEIPT',
+  OTHERS = 'OTHERS',
+}
+
+/**
+ * Document period
+ */
+export interface DocumentPeriod {
+  count: number;
+  unit: DocumentUnitPeriod;
+}
+
+/**
+ * Unit period of document
+ */
+export enum DocumentUnitPeriod {
+  YEAR = 'YEAR',
+  MONTH = 'MONTH',
+  WEEK = 'WEEK',
+  DAY = 'DAY',
+}
+
+/**
+ * Legal file type
+ */
+export enum LegalFileType {
+  CONTRACT = 'CONTRACT',
+  INFORMATION = 'INFORMATION',
+  CONVENTION_OF_PROOF = 'CONVENTION_OF_PROOF',
+  LOAN_AND_PAYMENT_AGREEMENT = 'LOAN_AND_PAYMENT_AGREEMENT',
+  TERMS_OF_SALE = 'TERMS_OF_SALE',
+  SEPA = 'SEPA',
+  BANK_RECORDS = 'BANK_RECORDS',
+  LEGAL_MENTIONS = 'LEGAL_MENTIONS',
+  RECEIPT = 'RECEIPT',
+  OTHERS = 'OTHERS',
+}
+
+/**
+ * State of the document
+ */
+export enum DocumentState {
+  CREATED,
+  IN_PROGRESS,
+  REJECTED,
+  SUSPECTED,
+  OK,
+}
+
+/**
+ * Category of legal document
+ */
+export enum LegalDocumentCategory {
+  INSURANCE_CONTRACT = 'INSURANCE_CONTRACT',
+  PRODUCT_CONTRACT = 'PRODUCT_CONTRACT',
+  DEBIT_MANDATE = 'DEBIT_MANDATE',
+  RECORDS = 'RECORDS',
+  LEGAL_MENTIONS = 'LEGAL_MENTIONS',
+  INFORMATION = 'INFORMATION',
+  RECEIPT = 'RECEIPT',
+  OTHERS = 'OTHERS',
+}
+
+/**
+ * Signature
+ */
+export interface ISignature {
+  callbackUrl: string;
+  createdAt: Date;
+  holder: Holder;
+  id: string;
+  legalDocumentIds: string[];
+  metadata: unknown;
+  partnerId: string;
+  redirectUrl: string;
+  redirectUrlCreatedAt: Date;
+  redirectUrlTTL: number;
+  state: SignatureState;
+}
+
+/**
+ * State of the signature
+ */
+export enum SignatureState {
+  CREATED,
+  IN_PROGRESS,
+  REJECTED,
+  // eslint-disable-next-line no-magic-numbers
+  SIGNED = 4,
```

Comment:
It looks weird that `3`  is not used but it is defined like that in the documentation.

Reply:
Again, it's just to be consistent with other states